spec/FieldsArray.ts

```diff
@@ -40,6 +40,15 @@ describe("FieldsArray", () => {
 			expect(fa[0]).toEqual({ key: "t1", value: "v1" });
 		});
 
+		it("should preserve order of input items when adding fields", () => {
+			expect(
+				fa.push({ key: "t1", value: "v1" }, { key: "t2", value: "v2" })
```

Comment:
I guess this is good, but that might be the case to extend the test also for splice... and maybe pop? I can do it